src/caller.jl

```diff
@@ -192,6 +192,18 @@ function create_ring_from_singular_ring(r::libSingular.ring_ptr)
       minpoly = F(libSingular.algExt_GetMinpoly(c, F.ptr))
       basering = N_AlgExtField(libSingular.nCopyCoeff(c), minpoly)
       T = n_algExt
+   elseif libSingular.nCoeff_is_Nemo_Field(c)
+      cf = libSingular.nCopyCoeff(c)
+      data_ptr = libSingular.nGetCoeffData(cf)
+      R = unsafe_pointer_to_objref(data_ptr)
+      basering = N_Field(R)
+      T = Nemo.Field
+   elseif libSingular.nCoeff_is_Nemo_Ring(c)
+      cf = libSingular.nCopyCoeff(c)
+      data_ptr = libSingular.nGetCoeffData(cf)
+      R = unsafe_pointer_to_objref(data_ptr)
+      basering = N_Ring(R)
+      T = Nemo.Ring
```

Comment:
Do you maybe want
```suggestion
      T = typeof(R)
```
here?

What exactly is the problem you are running into? I created a simple Singular polynomial ring over a Nemo polynomial ring as follows:
```
julia> F = Nemo.fraction_field(Nemo.polynomial_ring(Nemo.ZZ)[1])
Fraction field
  of univariate polynomial ring in x over ZZ

julia> R, x = polynomial_ring(F, [:x])
(Singular polynomial ring (Coeffs(22)),(x),(dp(1),C), spoly{Singular.n_FieldElem{AbstractAlgebra.Generic.FracFieldElem{Nemo.ZZPolyRingElem}}}[x])

julia> typeof(R)
PolyRing{Singular.n_FieldElem{AbstractAlgebra.Generic.FracFieldElem{Nemo.ZZPolyRingElem}}}
```


Now I guess I need to call some Singular library function which takes this ring and produces a new one.. or so... and then something goes wrong(?). Perhaps you could provide an example snippet doing such a call into Singular which show cases the issue?

(Clearly this would also be a a good starting point for adding a test case)

Reply:
applications are definitions of quotient rings non-commutative rings.
An artificial example is
julia> S = Singular.create_ring_from_singular_ring(Singular.libSingular.rCopy(R.ptr))
ERROR: MethodError: no method matching Singular.N_Field(::AbstractAlgebra.Generic.FracField{ZZPolyRingElem})
Stacktrace:
 [1] create_ring_from_singular_ring(r::CxxWrap.CxxWrapCore.CxxPtr{Singular.libSingular.ring})
   @ Singular ~/git/newSingularjl/src/caller.jl:199
 [2] top-level scope
   @ REPL[12]:1

Reply:
compare to
julia> r, (x, y) = Singular.polynomial_ring(Singular.QQ, ["x", "y"])
(Singular polynomial ring (QQ),(x,y),(dp(2),C), spoly{n_Q}[x, y])
julia> SS = Singular.create_ring_from_singular_ring(Singular.libSingular.rCopy(r.ptr))
Singular polynomial ring (QQ),(x,y),(dp(2),C)